card/mod/pointer/set/abstract/02_pointer/item_api.rb

```diff
@@ -0,0 +1,162 @@
+# ~~~~~~~~~~~~ READING ITEMS ~~~~~~~~~~~~
+
+# While each of the three main methods for returning lists of items can handle arguments,
+# they are most commonly used without them.
+
+# @return [Array] list of Card::Name objects
+# @param args [Hash]
+# @option args [String] :content override card content
+# @option args [String, Card::Name, Symbol] :context name in whose context relative items
+#         will be interpreted. For example. +A in context of B is interpreted as B+A
+#         context defaults to pointer card's name. If value is `:raw`, then name is not
+#         contextualized
+# @option args [String, Integer] :limit max number of cards to return
+# @option args [String, Integer] :offset begin after the offset-th item
+def item_names args={}
+  raw_item_strings(args[:content], args[:limit], args[:offset]).map do |item|
+    clean_item_name item, args[:context]
+  end.compact
+end
+
+# @return [Array] list of integers (card ids of items)
+# @param args [Hash] see #item_names
+def item_ids args={}
+  item_names(args).map { |name| Card.fetch_id name }.compact
+end
+
+# @return [Array] list of Card objects
+# @param args [Hash] see #item_names for additional options
+# @option args [String] :complete keyword to use in searching among items
+# @option args [True/False] :known_only if true, return only known cards
+# @option args [String] :type name of type to be used for unknown cards
+def item_cards args={}
+  return item_cards_search(args) if args[:complete]
+  return known_item_cards(args) if args[:known_only]
+  all_item_cards args
+end
+
+# ~~~~~~~~~~~~ ALTERING ITEMS ~~~~~~~~~~~~
```

Comment:
if you need comments to organize a file you probably should split it in several files/modules

Reply:
not sure I agree. It would be fine to move the "altering" ones to another module. But, while I wanted to make a distinction between strongly and weakly supported API (and I think comments are appropriate for that), I definitely think it would be more confusing to have the "weakly" supported helper methods in a separate file.

fwiw, this file has well under a hundred lines of code now; it's just the documentation that makes it seem long.